src/main/scala/uk/gov/nationalarchives/tdr/api/consignmentstatevalidation/ConsignmentStateValidator.scala

```diff
@@ -0,0 +1,43 @@
+package uk.gov.nationalarchives.tdr.api.consignmentstatevalidation
+
+import sangria.execution.{BeforeFieldResult, Middleware, MiddlewareBeforeField, MiddlewareQueryContext}
+import sangria.schema.Context
+import uk.gov.nationalarchives.tdr.api.graphql.ConsignmentApiContext
+
+import scala.concurrent.duration._
+import scala.concurrent.{Await, ExecutionContext}
+
+class ConsignmentStateValidator (implicit executionContext: ExecutionContext)
+  extends Middleware[ConsignmentApiContext] with MiddlewareBeforeField[ConsignmentApiContext] {
+
+  override type QueryVal = Unit
+  override type FieldVal = Unit
+
+  override def beforeQuery(context: MiddlewareQueryContext[ConsignmentApiContext, _, _]): QueryVal = ()
+
+  override def afterQuery(queryVal: QueryVal, context: MiddlewareQueryContext[ConsignmentApiContext, _, _]): Unit = ()
+
+  override def beforeField(queryVal: QueryVal,
+                           mctx: MiddlewareQueryContext[ConsignmentApiContext, _, _],
+                           ctx: Context[ConsignmentApiContext, _]
+                          ): BeforeFieldResult[ConsignmentApiContext, Unit] = {
+    val validationList: Seq[BeforeFieldResult[ConsignmentApiContext, Unit]] = ctx.field.tags.map {
+      case v: ConsignmentStateTag => {
+        val validationResult = v.validate(ctx)
+
+        // Awaiting a Future is risky because the thread will block until the response is returned or the timeout is reached.
+        // It could cause the API to be slow because akka-http cannot assign threads to new requests while this one is
+        // blocked.
+        //
+        // We are only using Await because Sangria middleware does not support Futures like the main resolvers do. We should
+        // remove it when we find a way to do authorisation in a completely async way in Sangria.
+        Await.result(validationResult, 5 seconds)
```

Comment:
This risky `Await.result` is now duplicated. It would be good to isolate it in one place, so that if we ever [fix it](https://national-archives.atlassian.net/browse/TDR-52), it will be fixed everywhere.

What do you think about moving the `Await` to the `validate` method of `Validation` tag? So it would look something like:

```
trait ValidationTag extends FieldTag {
  
  def validate(ctx: Context[ConsignmentApiContext, _])
              (implicit executionContext: ExecutionContext): BeforeFieldResult[ConsignmentApiContext, Unit] = {
    val validationResult = validate(ctx)

    // Comment about why this is bad
    Await.result(validationResult, 5 seconds)
  }

  def validateAsync(ctx: Context[ConsignmentApiContext, _])
              (implicit executionContext: ExecutionContext): Future[BeforeFieldResult[ConsignmentApiContext, Unit]]

  val continue: BeforeFieldResult[ConsignmentApiContext, Unit] = BeforeFieldResult(())
}
```

Reply:
Implemented this so validate function is in one place.